packages/marketing-modal/src/MarketingModal/MarketingModal.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import Button from '@leafygreen-ui/button';
+import { H3, Link } from '@leafygreen-ui/typography';
+import Modal from '@leafygreen-ui/modal';
+import { CloseIconColor } from '@leafygreen-ui/modal';
+import { svgBlobs } from '..';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import {
+  baseModalStyle,
+  baseGraphicContainerStyle,
+  buttonStyle,
+  centeredGraphicContainerStyle,
+  filledGraphicContainerStyle,
+  baseGraphicStyle,
+  filledGraphicStyle,
+  contentStyle,
+  contentThemeStyle,
+  titleStyle,
+  footerContentStyle,
+  linkStyle,
+  wrapperStyle,
+} from './styles';
+import { palette } from '@leafygreen-ui/palette';
+import { BlobPosition, GraphicStyle, MarketingModalProps } from './types';
+
+export const renderCurvedSVG = (darkMode: boolean) => {
+  const curvedSVGStyles = css`
+    position: absolute;
+    left: 0;
+    bottom: 24px;
+    color: ${darkMode ? palette.black : '#ffffff'};
+  `;
+
+  return (
+    <svg
+      className={cx(curvedSVGStyles)}
+      viewBox="0 0 600 49"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <path
+        d="M329.065 48C439.779 45.2633 537.038 27.0233 600 3.86855e-06V49H0V0C62.9624 27.0233 160.221 45.2633 270.935 48H329.065Z"
+        fill="currentColor"
+      />
+    </svg>
+  );
+};
+
+const MarketingModal = ({
+  children,
+  title,
+  graphic,
+  graphicStyle = GraphicStyle.Center,
+  onButtonClick,
+  onLinkClick,
+  onClose,
+  buttonText,
+  linkText,
+  darkMode: darkModeProp,
+  closeIconColor = CloseIconColor.Dark,
+  blobPosition = BlobPosition.TopLeft,
+  showBlob = false,
+  ...modalProps
+}: MarketingModalProps) => {
+  const { theme, darkMode } = useDarkMode(darkModeProp);
+
+  return (
+    <Modal
+      {...modalProps}
+      contentClassName={baseModalStyle}
+      setOpen={onClose}
```

Comment:
should this be passed to shouldClose instead?

Reply:
Hm, It shouldn't be passed to `shouldClose` since that returns a boolean that determines *if* the modal should close when someone tries to close it. From the types it looks like `onClose` doesn't return anything. In the [Story](https://github.com/mongodb/leafygreen-ui/blob/main/packages/marketing-modal/src/MarketingModal.story.tsx#L75) it looks like `onClose` is used to change the state of `setOpen`. So it looks like this controls closing the modal as well as allowing the user to add a callback. It's confusing tbh.